plugins/woocommerce/client/admin/client/settings-payments/onboarding/providers/woopayments/data/onboarding-context.tsx

```diff
@@ -120,11 +126,27 @@ export const OnboardingProvider: React.FC< {
 		( stepKey: string ) => {
 			const step = getStepByKey( stepKey );
 			if ( step?.path ) {
-				const newPath = getNewPath( { path: step.path }, step.path, {
-					page: 'wc-settings',
-					tab: 'checkout',
-				} );
-				history.push( newPath );
+				if ( source === 'settings-payments' ) {
```

Comment:
Another solution here, other than using the source, is to pass a function as a prop to take care of navigation. The function will get the `step` as an argument, and then decide the format of the new link (or even take care of the navigation itself 🤔 )

WDYT @mordeth ?